src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/createExpandedRowHeaders.test.ts

```diff
@@ -0,0 +1,1549 @@
+          span: 1,
+          text: '3',
+        },
+      ],
+    ]);
+  });
+
+  test('should return multiple rows of row headers if groups are provided', () => {
+    const rowHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3').addChild(new Header('2.1', '2.1')),
+      new Header('4', '4').addChild(new Header('3.1', '3.1')),
+    ];
+
+    expect(createExpandedRowHeaders(rowHeaders)).toEqual([
+      [
+        {
+          crossSpan: 1,
+          id: '1',
+          isGroup: false,
+          span: 1,
+          text: '1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '2',
+          isGroup: false,
+          span: 1,
+          text: '2',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '3',
+          isGroup: true,
+          span: 1,
+          text: '3',
+        },
+        {
+          crossSpan: 1,
+          id: '2.1',
+          isGroup: false,
+          span: 1,
+          text: '2.1',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '4',
+          isGroup: true,
+          span: 1,
+          text: '4',
+        },
+        {
+          crossSpan: 1,
+          id: '3.1',
+          isGroup: false,
+          span: 1,
+          text: '3.1',
+        },
+      ],
+    ]);
+  });
+
+  describe('returns correct headers with multiple levels of headers', () => {
+    test('renders 2x2 table correctly', () => {
+      const rowHeaders: Header[] = [
+        new Header('1', '1')
+          .addChild(new Header('3', '3'))
+          .addChild(new Header('4', '4')),
+        new Header('2', '2')
+          .addChild(new Header('3', '3'))
+          .addChild(new Header('4', '4')),
+      ];
+      const expandedRowHeaders = [
+        [
+          {
+            id: '1',
+            text: '1',
+            span: 2,
+            isGroup: true,
+            crossSpan: 1,
+          },
+          {
+            id: '3',
+            text: '3',
+            span: 1,
+            isGroup: false,
+            crossSpan: 1,
+          },
+        ],
+        [
+          {
+            id: '4',
+            text: '4',
+            span: 1,
+            isGroup: false,
+            crossSpan: 1,
+          },
+        ],
+        [
+          {
+            id: '2',
+            text: '2',
+            span: 2,
+            isGroup: true,
+            crossSpan: 1,
+          },
+          {
+            id: '3',
+            text: '3',
+            span: 1,
+            isGroup: false,
+            crossSpan: 1,
+          },
+        ],
+        [
+          {
+            id: '4',
+            text: '4',
+            span: 1,
+            isGroup: false,
+            crossSpan: 1,
+          },
+        ],
+      ];
+
+      expect(createExpandedRowHeaders(rowHeaders)).toEqual(expandedRowHeaders);
+    });
+
+    test('renders 2x2x2 table correctly', () => {
```

Comment:
Similar to above, this isn't rendering a table so something like 'returns correct headers with 3 levels of 2 row headers' would be more appropriate.